### PR TITLE
Drop support for NaN categories in Categorical

### PR DESCRIPTION
**Pull request by gfyoung**
Deprecated in 0.17.0.

xref #10748
xref #13648

### CODE REVIEW

File: pandas/tests/test_categorical.py [hunk 1]
```diff
@@ -222,29 +216,19 @@ def f():
...
-                                 categories=[np.nan, "a", "b", "c"])
-        self.assertTrue(is_object_dtype(cat.categories))
-        # but don't do it for floats
-        with tm.assert_produces_warning(FutureWarning):
-            cat = pd.Categorical([np.nan, 1., 2., 3.],
-                                 categories=[np.nan, 1., 2., 3.])
-        self.assertTrue(is_float_dtype(cat.categories))
+
+        # Cannot have NaN in categories
+        def f():
```
Comment (jreback) [inline pandas/tests/test_categorical.py]: can you add a test for NaT as well, and FYI, you can use pytest syntax

```
with pytest.raises(ValueError):
    ....
```

**Comment by jreback** [comment]
xref #13648 testing with ``None`` as well.

File: pandas/tests/test_categorical.py (hunk 1 above)
Comment (gfyoung) [inline pandas/tests/test_categorical.py]: Done.

**Comment by gfyoung** [comment]
@jreback : Added tests for #13648, and everything is still green.

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**